usaspending_api/etl/management/commands/reindex_elastic.py

```diff
@@ -0,0 +1,157 @@
+                        },
+                "dest": {
+                        "index": dest
+                        }
+                }
+        logging.warn('CREATING INDEX {}'.format(dest))
+        ES_CLIENT.reindex(body=body_)
+
+    def indicies_dict(self):
+        index_name = self.source_prefix + '*'
+        source_indices = ES_CLIENT.indices.get(index_name).keys()
+        source_indices = sorted(source_indices, key=lambda x: x.split('-')[-1])
+
+        dest_indices = [i.replace(self.source_prefix, self.dest_prefix) for i in source_indices]
+        self.reindex_dict = dict(zip(source_indices, dest_indices))
+        count = self.get_total_count(index_name)
+        logging.warn('Reindexing a total of {} documents'.format(count))
+
+    def get_total_count(self, index_name):
+        try:
+            response = ES_CLIENT.search(index=index_name)
+            return response["hits"]["total"]
+        except TransportError as e:
+            logging.error('[CONNECTION TIMEOUT] with ElasticSearch cluster: {e}'.
+                          format(e=str(e)))
+            return -1
+
+    def verify(self, source, target):
+        difference = self.get_total_count(source) - self.get_total_count(target)
+        if difference != 0:
+            logging.warn('Still have {} documents to reindex'.format(difference))
+        return difference, difference/INGEST_RATE
+
+    def controller(self):
+        logging.warn('Reindexing the following \n')
+        pprint(self.reindex_dict)
+        print('\n\n')
+        for source, dest in self.reindex_dict.items():
+            if ES_CLIENT.indices.exists(dest):
+                if not self.wipe:
+                    logging.warn('{} exists... passing ... '.format(dest))
+                else:
+                    logging.warn('{} exists... deleting... {}'.format(dest))
+                    ES_CLIENT.indices.delete(dest)
+            if not ES_CLIENT.indices.exists(dest):
+                self.size, self.sleep_time = self.verify(source, dest)
+                logging.warn('Reindexing {} -- {} documents'.format(source, self.size))
+                try:
+                    self.single_reindex(source, dest)
+                    logging.info(source)
+                    ES_CLIENT.indices.flush('*')
+                except (ConnectionTimeout) as e:
+                    logging.error('[CONNECTION TIMEOUT] with ElasticSearch cluster: {e}'.
+                                  format(e=str(e)))
+                    for attempt in range(MAX_RETRIES):
+                        self.size, self.sleep_time = self.verify(source, dest)
+                        if self.sleep_time > 0:
+                            logging.warn('---------- Sleep_time of index >>> {} ------------------'
+                                         .format(str(self.sleep_time)))
+                            time.sleep(self.sleep_time)
+        print('\n\n')
+        index_name = self.dest_prefix + '*'
+        ES_CLIENT.indices.flush('*')
```

Comment:
Would it be better to flush only the indices being modified in this script?

Reply:
non-trivial

Reply:
It is non-trival to change the code to flush only the touched indexes?
Or the impact of flushing all is trivial? 

If the latter, is there a possibility this will no longer hold true? Would more data/more indexes make this expensive at times?

Reply:
Its not expensive- it'd only be expensive on the first call. It's to try to maintain performance in the background while this is going on.

Reply:
So to summarize:
Calling flush on the entire cluster is an expensive operation the first time it is called. Subsequent calls are trival.
Correct?

It appears this is only called once in the script.